plugin/src/main/java/org/autorefactor/refactoring/Refactorings.java

```diff
@@ -413,4 +418,40 @@ public BadLocationException call() throws Exception {
             throw ex;
         }
     }
+
+    /**
+     * Gets the ASTRewrite rewrite object.
+     *
+     * @return the ASTRewrite rewrite
+     */
+    public ASTRewrite getRewrite() {
+        return rewrite;
+    }
+
+    /**
+     * Insert a statement in the end of a method body.
+     *
+     * @param stmt the statement to be added
+     * @param md the method declaration
+     */
+    public void insertLast(Statement stmt, MethodDeclaration md) {
+        insertAt(stmt,
+                md.getBody().statements().size(),
+                Block.STATEMENTS_PROPERTY,
+                md.getBody());
+    }
+
+    /**
+     * Insert a declaration in the end of a type declaration body.
+     *
+     * @param declaration the declaration to be added
+     * @param td the type declaration
+     */
+    public void insertLast(BodyDeclaration declaration, TypeDeclaration td) {
+        insertAt(
+                declaration,
+                td.bodyDeclarations().size(),
+                td.getBodyDeclarationsProperty(),
+                td);
+    }

```

Comment:
This is not what I expected :)

I am trying to write code as generic as possible here.
These two methods are very specific.

I was more expecting something like this:
```java
    public void insertLast(ASTNode nodeToInsert, StructuralPropertyDescriptor locationInParent, ASTNode listHolder) {
        getListRewrite(listHolder, locationInParent).insertLast(nodeToInsert, null);
        addRefactoredNodes(listHolder);
    }
```
I am going to commit such method anyway because we will certainly need them in the future.

That being said I can see where you are going and it makes sense to me.
I am considering my options for handling this :)